### ISSUE TITLE
Message to self never returns

### ISSUE THREAD

**Issue by twschiller** [pixiebrix/pixiebrix-extension#1542]
- In preload, the background script was calling the ensureContextMenu via the messenger API
- It would get stuck in retry
- It should just run the handler directly without sending a message (potentially warning about an unnecessary message)

**Comment by fregante** [comment]
See https://github.com/pixiebrix/pixiebrix-extension/issues/1542

I think this is a bad pattern. If there's a local function it should be called directly. If we're back to sharing the same exact code in both contexts we're back to square one IMHO

**Comment by fregante** [comment]
Looking at how the Point was implemented (i.e. explicitly to be shared across contexts), I think this might still be necessary.

The difference is that most lifted functions did not have access to the sender before, and now they will, and the sender will be `null`

**Comment by twschiller** [comment]
> I think this is a bad pattern

I agree it's probably better to have context-aware callers and have the component explicitly decide whether to call the concrete method or the API version. IIRC, contextMenu is an outlier since the background page is what gets the initial event to trigger an extension

I wonder how to make incorrect usage more obvious during testing as we switch things over? Perhaps using warning/error colors in the retry logic logging if the message hasn't been handled after _X_ seconds?

**Comment by fregante** [comment]
Yes definitely it should be handled somehow. Depending on how common this type of shared component is, we can decide whether to support it (I was writing something just earlier) or not. 

The alternative I suggested is actually worse as far as weight is concerned because it would require importing both the messenger/api and the native version:

``` JavaScript 
import * as api from "@/contentScript/api"
import { contextMenuThing } from "@/contextMenu/thing"

isBackground()?contextMenuThing() : api.contextMenuThing()
``` 

Webpack might or might not treeshake this.

**Comment by fregante** [comment]
POC to fix this issue in https://github.com/pixiebrix/webext-messenger/pull/33#pullrequestreview-768091252

Let me know how you'd like to proceed.
